pan/strings.pan

```diff
@@ -0,0 +1,34 @@
+@documentation{
+Convenience string types intended for use in building up more complex types.
+}
+declaration template pan/strings;
+
+
+@documentation{
+A string that must contain at least one character (anything other than an empty string).
+}
+type string_non_empty = string(1..);
+
+
+@documentation{
+A string that must contain at least one character and may not contain any whitespace.
+}
+type string_non_whitespace = string_non_empty with !match(SELF, '\s');
```

Comment:
`with match(SELF, '\S')`?

Reply:
No, that would mean "at least one non-whitespace character"

Reply:
you are right! I wanted to say: `with match(SELF, '^\S+$')` . But probably yours is more efficient, agreed.